# Exploring the combined data for O'hare for the year 2010 - 2019.

### CHICAGO OHARE INTERNATIONAL AIRPORT, IL US

ID	GHCND:USW00094846  
Lat/Lon:	41.995, -87.9336

### Data source

https://www.ncdc.noaa.gov/cdo-web/search

### Data

WT03 - Thunder  
WT04 - Ice pellets, sleet, snow pellets, or small hail"  
**PRCP - Precipitation**  
WT05 - Hail (may include small hail)  
WT06 - Glaze or rime  
WT07 - Dust, volcanic ash, blowing dust, blowing sand, or blowing obstruction  
WT08 - Smoke or haze  
**SNWD - Snow depth**  
WT09 - Blowing or drifting snow  
WDF2 - Direction of fastest 2-minute wind  
WDF5 - Direction of fastest 5-second wind  
PGTM - Peak gust time  
WT11 - High or damaging winds  
**TMAX - Maximum temperature**  
WT13 - Mist  
WSF2 - Fastest 2-minute wind speed  
FMTM - Time of fastest mile or fastest 1-minute wind  
WSF5 - Fastest 5-second wind speed  
**SNOW - Snowfall**  
WT14 - Drizzle  
WT15 - Freezing drizzle  
WT16 - Rain (may include freezing rain, drizzle, and freezing drizzle)"  
WT17 - Freezing rain  
WT18 - Snow, snow pellets, snow grains, or ice crystals  
WT19 - Unknown source of precipitation  
**AWND - Average wind speed**  
WT21 - Ground fog  
WT22 - Ice fog or freezing fog  
WT01 - Fog, ice fog, or freezing fog (may include heavy fog)  
WESD - Water equivalent of snow on the ground  
WT02 - Heavy fog or heaving freezing fog (not always distinguished from fog)  
**TAVG - Average Temperature**  
**TMIN - Minimum temperature**  

In [19]:
import pandas as pd
import time

In [20]:
data = pd.read_csv('../data/ohare/all_ohare_data_2010_2019.csv')
print(data.shape)
data.head()

(3377, 36)


STATION                                        NAME        DATE   AWND  \
0  USW00094846  CHICAGO OHARE INTERNATIONAL AIRPORT, IL US  2010-01-01  10.29   
1  USW00094846  CHICAGO OHARE INTERNATIONAL AIRPORT, IL US  2010-01-02  11.86   
2  USW00094846  CHICAGO OHARE INTERNATIONAL AIRPORT, IL US  2010-01-03  10.29   
3  USW00094846  CHICAGO OHARE INTERNATIONAL AIRPORT, IL US  2010-01-04  11.41   
4  USW00094846  CHICAGO OHARE INTERNATIONAL AIRPORT, IL US  2010-01-05   8.50   

     FMTM    PGTM  PRCP  SNOW  SNWD  TAVG  ...  WT11  WT13  WT14  WT15  WT16  \
0  2359.0  2351.0   0.0   0.0   2.0   NaN  ...   NaN   NaN   NaN   NaN   NaN   
1   928.0  2206.0   0.0   0.0   2.0   NaN  ...   NaN   NaN   NaN   NaN   NaN   
2  2114.0  1917.0   0.0   0.0   2.0   NaN  ...   NaN   NaN   NaN   NaN   1.0   
3  1644.0  1300.0   0.0   0.0   2.0   NaN  ...   NaN   NaN   NaN   NaN   1.0   
4  1518.0  1500.0   0.0   0.0   2.0   NaN  ...   NaN   NaN   NaN   NaN   1.0   

   WT17  WT18  WT19  WT21  WT22  
0   NaN   1.0   NaN   NaN   NaN  
1   NaN   NaN   NaN   NaN   NaN  
2   NaN   1.0   NaN   NaN   NaN  
3   NaN   1.0   NaN   NaN   NaN  
4   NaN   1.0   NaN   NaN   NaN  

[5 rows x 36 columns]

In [21]:
data.dtypes

STATION     object
NAME        object
DATE        object
AWND       float64
FMTM       float64
PGTM       float64
PRCP       float64
SNOW       float64
SNWD       float64
TAVG       float64
TMAX         int64
TMIN         int64
WDF2         int64
WDF5       float64
WESD       float64
WSF2       float64
WSF5       float64
WT01       float64
WT02       float64
WT03       float64
WT04       float64
WT05       float64
WT06       float64
WT07       float64
WT08       float64
WT09       float64
WT11       float64
WT13       float64
WT14       float64
WT15       float64
WT16       float64
WT17       float64
WT18       float64
WT19       float64
WT21       float64
WT22       float64
dtype: object

In [22]:
data.isnull().sum()

STATION       0
NAME          0
DATE          0
AWND          0
FMTM       2739
PGTM       2743
PRCP          0
SNOW          0
SNWD          0
TAVG       1186
TMAX          0
TMIN          0
WDF2          0
WDF5         13
WESD       3012
WSF2          0
WSF5         13
WT01       2178
WT02       3277
WT03       3039
WT04       3334
WT05       3268
WT06       3342
WT07       3357
WT08       2745
WT09       3337
WT11       3368
WT13       3008
WT14       3299
WT15       3373
WT16       2911
WT17       3372
WT18       3182
WT19       3376
WT21       3374
WT22       3359
dtype: int64

In [23]:
features = ['DATE', 'AWND', 'PRCP', 'SNOW', 'SNWD', 'TAVG', 'TMAX', 'TMIN', 'WDF2', 'WSF2']

In [24]:
data = data[features]
print(data.shape)
data.head()

(3377, 10)


DATE   AWND  PRCP  SNOW  SNWD  TAVG  TMAX  TMIN  WDF2  WSF2
0  2010-01-01  10.29   0.0   0.0   2.0   NaN    16     5   320  15.0
1  2010-01-02  11.86   0.0   0.0   2.0   NaN    11     2   340  17.9
2  2010-01-03  10.29   0.0   0.0   2.0   NaN    18    -1   330  16.1
3  2010-01-04  11.41   0.0   0.0   2.0   NaN    18     7   310  17.0
4  2010-01-05   8.50   0.0   0.0   2.0   NaN    23    13   340  16.1

In [25]:
data.SNWD.value_counts()

0.0     2951
2.0       74
1.2       57
5.1       40
1.0       35
3.0       32
3.1       28
3.9       26
5.9       21
4.0       14
9.1       13
5.0       13
7.1       10
9.8        9
6.0        7
7.9        6
11.8       5
11.0       5
17.0       4
7.0        3
18.0       3
14.2       3
10.0       2
13.0       2
8.0        2
9.0        2
15.0       2
19.0       2
16.1       1
16.9       1
21.0       1
16.0       1
18.1       1
14.0       1
Name: SNWD, dtype: int64

In [26]:
data.isnull().sum()

DATE       0
AWND       0
PRCP       0
SNOW       0
SNWD       0
TAVG    1186
TMAX       0
TMIN       0
WDF2       0
WSF2       0
dtype: int64

In [27]:
data.WDF2.max()

360

In [28]:
data.WDF2.min()

10

In [29]:
data.WDF2.value_counts()

40     159
210    154
200    151
280    140
20     140
320    133
190    133
250    127
300    125
330    123
50     121
260    120
100    120
340    116
270    104
290     99
310     98
220     97
240     94
180     93
30      92
60      82
70      79
170     76
10      68
230     63
110     60
350     60
160     49
120     48
90      47
130     46
360     45
140     42
80      40
150     33
Name: WDF2, dtype: int64

In [30]:
data.DATE

0       2010-01-01
1       2010-01-02
2       2010-01-03
3       2010-01-04
4       2010-01-05
5       2010-01-06
6       2010-01-07
7       2010-01-08
8       2010-01-09
9       2010-01-10
10      2010-01-11
11      2010-01-12
12      2010-01-13
13      2010-01-14
14      2010-01-15
15      2010-01-16
16      2010-01-17
17      2010-01-18
18      2010-01-19
19      2010-01-20
20      2010-01-21
21      2010-01-22
22      2010-01-23
23      2010-01-24
24      2010-01-25
25      2010-01-26
26      2010-01-27
27      2010-01-28
28      2010-01-29
29      2010-01-30
           ...    
3347    2019-03-02
3348    2019-03-03
3349    2019-03-04
3350    2019-03-05
3351    2019-03-06
3352    2019-03-07
3353    2019-03-08
3354    2019-03-09
3355    2019-03-10
3356    2019-03-11
3357    2019-03-12
3358    2019-03-13
3359    2019-03-14
3360    2019-03-15
3361    2019-03-16
3362    2019-03-17
3363    2019-03-18
3364    2019-03-19
3365    2019-03-20
3366    2019-03-21
3367    2019-03-22
3368    2019

### Note: there are 3377 days between 01-01-2010/03-31-2019, when you include both the first and last days in the set.

In [42]:
# there are no empty values, and every day is accounted for during this datetime interval
data[data.TAVG.notnull()]

DATE   AWND  PRCP  SNOW  SNWD  TAVG  TMAX  TMIN  WDF2  WSF2
1186 2013-04-01  11.63  0.00   0.0   0.0  36.0    41    29   310  21.9
1187 2013-04-02   8.95  0.00   0.0   0.0  34.0    45    25   320  21.0
1188 2013-04-03   6.93  0.00   0.0   0.0  35.0    44    25    60  14.1
1189 2013-04-04   8.05  0.00   0.0   0.0  39.0    58    25   240  16.1
1190 2013-04-05  11.86  0.00   0.0   0.0  44.0    45    36    20  21.0
1191 2013-04-06  18.57  0.04   0.0   0.0  47.0    70    36   180  36.9
1192 2013-04-07  11.41  0.00   0.0   0.0  53.0    59    40   270  25.1
1193 2013-04-08  13.87  0.49   0.0   0.0  50.0    64    41   260  33.1
1194 2013-04-09  11.86  0.20   0.0   0.0  44.0    55    39    20  23.0
1195 2013-04-10  16.11  0.94   0.0   0.0  40.0    41    38   290  33.1
1196 2013-04-11   9.40  0.24   0.0   0.0  41.0    46    38    70  25.9
1197 2013-04-12  13.42  0.00   0.0   0.0  41.0    45    37   260  23.9
1198 2013-04-13  12.53  0.00   0.0   0.0  39.0    45    34   270  23.9
1199 2013-04-14  14.99  0.00   0.0   0.0  45.0    69    38   190  30.0
1200 2013-04-15  13.42  0.36   0.0   0.0  61.0    66    44   200  32.0
1201 2013-04-16   7.61  0.01   0.0   0.0  47.0    58    40   110  15.0
1202 2013-04-17  10.96  2.01   0.0   0.0  43.0    49    42    70  25.9
1203 2013-04-18  15.43  3.54   0.0   0.0  52.0    63    39   170  33.1
1204 2013-04-19  19.46  0.01   0.1   0.0  41.0    40    33   250  32.0
1205 2013-04-20   9.40  0.00   0.0   0.0  37.0    47    31   310  19.9
1206 2013-04-21  11.41  0.00   0.0   0.0  40.0    54    29   150  19.9
1207 2013-04-22   9.62  0.00   0.0   0.0  51.0    67    38   130  18.1
1208 2013-04-23   9.62  0.70   0.0   0.0  53.0    59    38   340  18.1
1209 2013-04-24  12.30  0.04   0.0   0.0  41.0    50    37   280  18.1
1210 2013-04-25  11.86  0.07   0.0   0.0  45.0    51    38   240  23.0
1211 2013-04-26  11.63  0.00   0.0   0.0  49.0    67    34   180  25.1
1212 2013-04-27   6.71  0.00   0.0   0.0  57.0    68    43   110  16.1
1213 2013-04-28   5.82  0.03   0.0   0.0  58.0    65    49    60  10.1
1214 2013-04-29  12.08  0.00   0.0   0.0  60.0    74    47   170  23.9
1215 2013-04-30  17.22  0.00   0.0   0.0  70.0    85    61   200  29.1
...         ...    ...   ...   ...   ...   ...   ...   ...   ...   ...
3347 2019-03-02   8.50  0.00   0.0   0.0  29.0    33    25   350  16.1
3348 2019-03-03  12.08  0.00   0.0   0.0  22.0    25     3   350  19.9
3349 2019-03-04  14.99  0.00   0.0   0.0   5.0    12    -3   250  21.9
3350 2019-03-05  15.43  0.00   0.2   0.0  12.0    21     8   290  23.0
3351 2019-03-06  11.86  0.00   0.0   0.0  16.0    30     9   260  21.9
3352 2019-03-07   6.04  0.00   0.0   0.0  23.0    28    14   320  13.0
3353 2019-03-08   5.37  0.00   0.1   0.0  28.0    39    21    80  10.1
3354 2019-03-09  14.09  0.56   0.0   0.0  33.0    43    26   250  29.1
3355 2019-03-10  19.69  0.00   0.0   0.0  37.0    38    29   250  31.1
3356 2019-03-11   9.62  0.00   0.0   0.0  32.0    41    27   280  16.1
3357 2019-03-12   9.17  0.00   0.0   0.0  35.0    49    24   180  21.0
3358 2019-03-13  11.86  0.54   0.0   0.0  45.0    58    40   150  25.1
3359 2019-03-14  20.36  0.40   0.0   0.0  55.0    62    39   240  40.9
3360 2019-03-15  14.99  0.00   0.0   0.0  38.0    40    32   240  25.1
3361 2019-03-16   8.50  0.00   0.0   0.0  34.0    41    28   300  15.0
3362 2019-03-17   7.61  0.00   0.0   0.0  35.0    43    28   290  18.1
3363 2019-03-18   7.61  0.00   0.0   0.0  35.0    43    27   270  17.0
3364 2019-03-19   7.83  0.00   0.0   0.0  39.0    49    30   210  15.0
3365 2019-03-20   8.50  0.14   0.0   0.0  42.0    44    38   260  18.1
3366 2019-03-21  11.18  0.00   0.0   0.0  40.0    45    36   350  17.0
3367 2019-03-22  10.96  0.00   0.0   0.0  40.0    46    29    30  23.9
3368 2019-03-23   4.47  0.00   0.0   0.0  36.0    53    25   180  12.1
3369 2019-03-24   8.72  0.05   0.0   0.0  45.0    50    34    20  25.9
3370 2019-03-25  17.45  0.00   0.0   0.0  36.0    42    26    20  28.0
3371 2019-03-26   4.92  0.00   0.0 

In [41]:
data[data.TAVG.notnull()].isnull().sum()

DATE    0
AWND    0
PRCP    0
SNOW    0
SNWD    0
TAVG    0
TMAX    0
TMIN    0
WDF2    0
WSF2    0
dtype: int64

In [38]:
data['DATE'] = pd.to_datetime(data['DATE'])
print(data.shape)
data.head()

(3377, 10)


DATE   AWND  PRCP  SNOW  SNWD  TAVG  TMAX  TMIN  WDF2  WSF2
0 2010-01-01  10.29   0.0   0.0   2.0   NaN    16     5   320  15.0
1 2010-01-02  11.86   0.0   0.0   2.0   NaN    11     2   340  17.9
2 2010-01-03  10.29   0.0   0.0   2.0   NaN    18    -1   330  16.1
3 2010-01-04  11.41   0.0   0.0   2.0   NaN    18     7   310  17.0
4 2010-01-05   8.50   0.0   0.0   2.0   NaN    23    13   340  16.1

In [40]:
data[data.TAVG.notnull()][(data['DATE'] >= '2015-01-01') & (data['DATE'] <= '2018-12-31')]

/Users/blakewallace/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.


DATE   AWND  PRCP  SNOW  SNWD  TAVG  TMAX  TMIN  WDF2  WSF2
1826 2015-01-01  14.54  0.00   0.0   0.0  20.0    32    14   240  23.9
1827 2015-01-02   4.70  0.00   0.0   0.0  26.0    34    15   150  12.1
1828 2015-01-03   5.59  0.57   0.4   0.0  31.0    34    29   140  13.0
1829 2015-01-04  15.88  0.13   2.2   1.2  29.0    33     2   320  25.9
1830 2015-01-05  10.29  0.15   1.8   2.0   2.0     6    -3   310  21.9
1831 2015-01-06  13.87  0.01   0.2   3.9   5.0    10    -5   270  23.9
1832 2015-01-07  17.45  0.00   0.0   3.1   3.0     6    -6   330  28.0
1833 2015-01-08  17.67  0.06   3.0   3.1  -2.0    18    -8   300  28.0
1834 2015-01-09  16.55  0.00   0.0   5.1   8.0    11     0   330  30.0
1835 2015-01-10  11.63  0.00   0.0   3.9   5.0    24    -2   180  21.9
1836 2015-01-11   8.28  0.19   2.6   3.9  26.0    32    22   180  18.1
1837 2015-01-12  12.75  0.00   0.4   5.9  23.0    28    15    10  21.0
1838 2015-01-13   7.61  0.00   0.2   5.9  17.0    19     3    50  19.9
1839 2015-01-14   6.49  0.00   0.0   5.9  11.0    19     1   220  14.1
1840 2015-01-15  10.07  0.00   0.0   5.1  22.0    34    17   240  21.0
1841 2015-01-16   8.28  0.00   0.0   3.9  28.0    34    23   140  15.0
1842 2015-01-17  15.43  0.00   0.0   3.9  34.0    42    26   210  28.0
1843 2015-01-18  10.51  0.00   0.0   0.0  37.0    40    27   300  19.9
1844 2015-01-19   5.37  0.00   0.0   0.0  32.0    41    25   100  12.1
1845 2015-01-20   7.38  0.00   0.0   0.0  34.0    36    29    40  17.0
1846 2015-01-21   7.61  0.01   0.1   0.0  31.0    33    29    20  14.1
1847 2015-01-22   6.93  0.00   0.0   0.0  30.0    31    28   260  15.0
1848 2015-01-23  11.41  0.00   0.0   0.0  30.0    31    29   220  17.0
1849 2015-01-24   8.05  0.00   0.0   0.0  33.0    43    29   260  18.1
1850 2015-01-25  14.54  0.08   0.6   0.0  31.0    34    24    40  25.9
1851 2015-01-26   5.59  0.07   1.0   0.0  23.0    26    15    60  13.0
1852 2015-01-27   6.04  0.00   0.0   1.2  27.0    32    20    40  13.0
1853 2015-01-28  10.96  0.01   0.0   1.2  26.0    34    19   170  19.9
1854 2015-01-29  13.87  0.01   0.1   1.2  33.0    35    28   320  25.9
1855 2015-01-30   7.38  0.00   0.0   1.2  27.0    30    18   350  19.9
...         ...    ...   ...   ...   ...   ...   ...   ...   ...   ...
3257 2018-12-02   8.95  0.10   0.0   0.0  41.0    44    35   340  18.1
3258 2018-12-03  12.97  0.01   0.3   0.0  34.0    35    29   360  23.9
3259 2018-12-04   7.83  0.00   0.0   0.0  31.0    32    28   320  14.1
3260 2018-12-05  10.51  0.00   0.0   0.0  28.0    33    25   250  16.1
3261 2018-12-06   9.84  0.00   0.0   0.0  31.0    33    19   320  17.0
3262 2018-12-07   6.26  0.00   0.0   0.0  21.0    25    16   230  18.1
3263 2018-12-08   3.80  0.00   0.0   0.0  24.0    32    20    20  10.1
3264 2018-12-09   4.25  0.00   0.0   0.0  24.0    35    17   350  12.1
3265 2018-12-10   6.93  0.00   0.0   0.0  23.0    34    16   220  15.0
3266 2018-12-11   7.61  0.00   0.0   0.0  30.0    38    25   230  16.1
3267 2018-12-12  10.74  0.00   0.0   0.0  34.0    41    29   230  21.9
3268 2018-12-13   4.25  0.15   0.0   0.0  33.0    46    24   160  14.1
3269 2018-12-14   6.71  0.13   0.0   0.0  39.0    40    32    10  13.0
3270 2018-12-15   7.61  0.00   0.0   0.0  35.0    46    26    30  14.1
3271 2018-12-16   6.49  0.00   0.0   0.0  35.0    49    25   260  14.1
3272 2018-12-17   7.61  0.00   0.0   0.0  37.0    42    28   350  17.0
3273 2018-12-18   6.71  0.00   0.0   0.0  33.0    42    25   230  14.1
3274 2018-12-19  10.96  0.00   0.0   0.0  39.0    49    33   180  18.1
3275 2018-12-20   7.83  0.10   0.0   0.0  44.0    46    38   360  15.0
3276 2018-12-21  13.65  0.05   0.0   0.0  35.0    38    29   360  21.9
3277 2018-12-22   6.26  0.00   0.0   0.0  30.0    34    28   200  13.0
3278 2018-12-23  11.18  0.00   0.0   0.0  33.0    38    28   260  23.9
3279 2018-12-24   5.82  0.00   0.0   0.0  29.0    33    23   270  17.0
3280 2018-12-25   3.58  0.00   0.0   0.0  34.0    45    28   240  12.1
3281 2018-12-26   5.14  0.00   0.0 

## This is the data for ohare from April 1, 2013 t0 March 31, 2019.

In [43]:
data_apr012013_mar312019 = data[data.TAVG.notnull()]
print(data_apr012013_mar312019.shape)
data_apr012013_mar312019.head()

(2191, 10)


DATE   AWND  PRCP  SNOW  SNWD  TAVG  TMAX  TMIN  WDF2  WSF2
1186 2013-04-01  11.63   0.0   0.0   0.0  36.0    41    29   310  21.9
1187 2013-04-02   8.95   0.0   0.0   0.0  34.0    45    25   320  21.0
1188 2013-04-03   6.93   0.0   0.0   0.0  35.0    44    25    60  14.1
1189 2013-04-04   8.05   0.0   0.0   0.0  39.0    58    25   240  16.1
1190 2013-04-05  11.86   0.0   0.0   0.0  44.0    45    36    20  21.0

In [44]:
data_apr012013_mar312019.dtypes

DATE    datetime64[ns]
AWND           float64
PRCP           float64
SNOW           float64
SNWD           float64
TAVG           float64
TMAX             int64
TMIN             int64
WDF2             int64
WSF2           float64
dtype: object

In [48]:
# saving the data
data_apr012013_mar312019.to_csv('../data/ohare/apr012013_mar312019.csv', index=False)

## This is the data for ohare from January 1, 2015 t0 December 31, 2018.

In [47]:
data_2015_2018 = data[data.TAVG.notnull()][(data['DATE'] >= '2015-01-01') & (data['DATE'] <= '2018-12-31')]
print(data_2015_2018.shape)
data_2015_2018.head()

(1461, 10)


/Users/blakewallace/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.


DATE   AWND  PRCP  SNOW  SNWD  TAVG  TMAX  TMIN  WDF2  WSF2
1826 2015-01-01  14.54  0.00   0.0   0.0  20.0    32    14   240  23.9
1827 2015-01-02   4.70  0.00   0.0   0.0  26.0    34    15   150  12.1
1828 2015-01-03   5.59  0.57   0.4   0.0  31.0    34    29   140  13.0
1829 2015-01-04  15.88  0.13   2.2   1.2  29.0    33     2   320  25.9
1830 2015-01-05  10.29  0.15   1.8   2.0   2.0     6    -3   310  21.9

In [50]:
# saving the data for Jan 01, 2015 to Dec 31, 2018
data_2015_2018.to_csv('../data/ohare/jan012015_dec312018.csv', index=False)

In [52]:
# saving the data for Apr 01, 2013 to Dec 31, 2018
data[data.TAVG.notnull()][(data['DATE'] >= '2013-04-01') & (data['DATE'] <= '2018-12-31')].to_csv('../data/ohare/apr012013_dec312018.csv', index=False)

/Users/blakewallace/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  
